### PR TITLE
Fix resources exclusion in a multi-module project with git repo

### PR DESCRIPTION
**Pull request by radoslaw-panuszewski**
## What's changed?
Fixed the bug with resources in subproject being excluded from processing.

## What's your motivation?
I have a multi-module project like this:
```
.
├── src
│   └── main
│       └── resources
│           └── application-root.yml
└── subproject
    └── src
        └── main
            └── resources
                └── application-subproject.yml  
```
And I have a custom recipe to modify some YAML. While the `application-root.yml` is correctly modified, the `application-subproject.yml` is skipped. It is reproduced in the `resources in subproject committed to git are correctly processed` test case.

There is already a test case `rewriteRun applies recipe to a multi-project build` which aims to verify this situation, but it doesn't check the case when Git repo is available. There is a different code path in `OmniParser` for such case and it's not reached.

I believe that I've found a bug in the code of `DefaultProjectParser`. When the `OmniParser` is created for the root project, it is configured to exclude contents of its subprojects. But when it's created for the subproject, it's still configured to exclude the root project's subprojects, so also the subproject we want to parse.

## Anything in particular you'd like reviewers to focus on?
I've added the `Project` parameter to the `omniParser()`. Please verify that the updated method usages are correct.

## Additional changes
There was a typo in `dependencyRepositoriesDeclaredInSettings` test case which caused the test to fail. I fixed it: `nevVersion` -> `newVersion`

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [ ] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [ ] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Review by bgalek — APPROVED** [review]

**Comment by timtebeek** [comment]
@travisboettcher this PR takes a slightly different approach to what you suggested in https://github.com/openrewrite/rewrite-gradle-plugin/issues/252 ; Any thoughts to share here?

File: plugin/src/test/kotlin/org/openrewrite/gradle/RewriteRunTest.kt [hunk 1]
```diff
@@ -573,7 +635,7 @@ class RewriteRunTest : RewritePluginTest {
                 }
                 
                 dependencies {
-                    implementation("com.fasterxml.jackson.core:jackson-core:2.16.1")
+                    implementation("com.fasterxml.jackson.core:jackson-core:2.16.0")
```
Comment (timtebeek) [inline plugin/src/test/kotlin/org/openrewrite/gradle/RewriteRunTest.kt:638]: Why this change?
```suggestion
                    implementation("com.fasterxml.jackson.core:jackson-core:2.16.1")
```

Comment (radoslaw-panuszewski) [reply plugin/src/test/kotlin/org/openrewrite/gradle/RewriteRunTest.kt:638]: @timtebeek I believe the story is like this:
* the original test was written when the latest version of `jackson-core` was `2.16.0`. Back then it was passing only accidentally, because there was a typo in YAML recipe config (`nevVersion` instead of `newVersion`) and `UpgradeDependencyVersion` uses default of `latest.release` when the `newVersion` is not provided
* newer version of `jackson-core` was released (`2.16.1`) which caused the test to fail
* I noticed the test failing locally and I fixed the typo `nevVersion` -> `newVersion`
* on master it was "fixed" by changing `2.16.0` to `2.16.1` in the assertion
* I rebased on master and changed it back to `2.16.0` because it's the correct value expected by this test 😉

Comment (timtebeek) [reply plugin/src/test/kotlin/org/openrewrite/gradle/RewriteRunTest.kt:638]: Thanks for the explanation! Makes sense now indeed, and thanks for the fix

**Review by timtebeek — APPROVED** [review]
Thanks a lot for the detailed look & fix!
